### PR TITLE
Add an extra layer for deshuffled prediction + zeroing out

### PR DESCRIPTION
**Pull request by avv-va**

### CODE REVIEW

File: ultralytics/models/yolo/pose_seg/train.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def set_model_attributes(self):
 
     def get_validator(self):
         """Returns an instance of the PoseValidator class for validation."""
-        self.loss_names = 'box_loss', 'pose_loss', 'kobj_loss', 'cls_loss', 'dfl_loss', 'seg_obj', 'seg_cls'
+        self.loss_names = 'box_loss', 'pose_loss', 'kobj_loss', 'cls_loss', 'dfl_loss', 'seg_obj_sh', 'seg_obj_unsh', 'seg_cls'
```
Comment (romainthibaux) [inline ultralytics/models/yolo/pose_seg/train.py:52]: It's a bit odd to call them "sh" and "unsh". For example, "seg_obj_sh" could be trained on unshuffled images based on rasterized boxes. It's more of a sequence: "seg_obj_sh" is level 0, "seg_obj_unsh" is level 1.
They could be named that way, or they could be considered a single block of channels, with the code referring to subchannels 0 or 1.

File: ultralytics/models/yolo/pose_seg/val.py [hunk 2]
```diff
@@ -36,9 +36,10 @@ def process_seg_result(self, preds):
         Pi_list = preds[1][0]
         x_flat = preds[0]
         seg_offset = 4 + self.nc
-        seg_obj = x_flat[:, seg_offset : seg_offset + 1, :]
-        seg_logits = x_flat[:, seg_offset + 1 : seg_offset + 1 + self.seg_ch_num, :]
-        return seg_obj, seg_logits, Pi_list
+        seg_obj_sh = x_flat[:, seg_offset : seg_offset + 1, :]
```
Comment (romainthibaux) [inline ultralytics/models/yolo/pose_seg/val.py:39]: Unused

File: ultralytics/nn/tasks.py [hunk 3]
```diff
@@ -406,11 +406,11 @@ def rasterize_boxes(self, img, gt_bboxes, gt_cls, batch_idx):
         B, _, img_H, img_W = img.shape
         anchor_H, anchor_W = img_H // min_stride, img_W // min_stride
 
-        gt_bboxes = xywh2xyxy(gt_bboxes.mul_(torch.Tensor([anchor_H, anchor_W, anchor_H, anchor_W]).to(gt_bboxes.device))) # T, 4
+        gt_bboxes_A = xywh2xyxy(gt_bboxes.mul(torch.Tensor([anchor_H, anchor_W, anchor_H, anchor_W]).to(gt_bboxes.device))) # T, 4
```
Comment (romainthibaux) [inline ultralytics/nn/tasks.py:409]: What does "A" mean? Maybe `gt_bboxes_scaled`?

File: ultralytics/utils/loss.py [hunk 4]
```diff
@@ -544,28 +562,45 @@ def __init__(self, model):
...
         super().__init__(model)
 
     def __call__(self, preds, batch):
-        _, _, _, _, pred_seg_obj, _ = self.prepare_preds(preds)
-        target_seg_obj = batch['seg_objectness'].flatten(start_dim=2)
+        loss = torch.zeros(2, device=self.device)
+        preds_dict = self.prepare_preds(preds)
+        pred_seg_obj0, pred_seg_obj1 = preds_dict[self.PredE.SEG_OBJ0], preds_dict[self.PredE.SEG_OBJ1]        
+        batch_size = pred_seg_obj0.shape[0]
+        last_unshuffled_idx = batch['seg_objectness_unsh'].shape[0]
```
Comment (romainthibaux) [inline ultralytics/utils/loss.py:586]: It's not a last index, it's a size. A "last" index would be size-1.
Since the predictions are now called "obj0" and "obj1", the keys "seg_objectness_unsh" and "seg_objectness_sh" should probably change too.

File: ultralytics/nn/tasks.py [hunk 5]
```diff
@@ -406,11 +406,11 @@ def rasterize_boxes(self, img, gt_bboxes, gt_cls, batch_idx):
         B, _, img_H, img_W = img.shape
         anchor_H, anchor_W = img_H // min_stride, img_W // min_stride
 
-        gt_bboxes = xywh2xyxy(gt_bboxes.mul_(torch.Tensor([anchor_H, anchor_W, anchor_H, anchor_W]).to(gt_bboxes.device))) # T, 4
+        gt_bboxes_scaled = xywh2xyxy(gt_bboxes.mul(torch.Tensor([anchor_H, anchor_W, anchor_H, anchor_W]).to(gt_bboxes.device))) # T, 4
```
[the remark this answers is not in the document]
Comment (avv-va) [reply ultralytics/nn/tasks.py]: changed to w h w h
